### DISCUSSION TITLE
5.1.4版本，配置多目录存储commitlog，持续压测发现不同目录commitlog名字相同

### DISCUSSION THREAD

**Discussion by domyway** [apache/rocketmq#8858] [Q&A]
配置的目录是
/data,/data2,/data3,/data4
经过长时间压测后，发现不同的目录下，会有相同文件的commitlog，其中一个commitlog看起来是空的，但是文件名相同。请问这是个BUG吗，有没有风险？ 如果进程重启，对重复的commitlog进行load，是不是也是风险？

代码分析不知道对不对：

MultiPathMappedFileQueue.java，这个文件是多路径下用的创建的文件。tryCreateMappedFile实现里，会将磁盘占用较高（85%）的路径给筛掉，不在上面创建新commitlog文件。

但是可能在前一次的create map file的时候，已经把它加在了nextNextFilePath中去预创建了。
如果下一次paths有改变，就会创建到不同的路径下作为真实的commitlog，而之前nextNext创建的是一个假的没有人用的。

需要大家的帮助

**Comment by domyway** [comment]
![image](https://github.com/user-attachments/assets/0a733c54-eafa-463e-9da7-0abc82aa9e74)

看起来的情况是：
第一个预创建的path只有一个是 /data1，等到第二次实际要用的时候，path的路径有2个了，/data和/data1，给路由到/data了，然后就实际创建到/data了

**Comment by hexueyuan** [comment]
看代码确实存在这个问题，并且最新版本也没看到有解决。想问下你们测试的时候这个问题会影响使用吗？看逻辑MappedFileQueue应该会自动跳过这种异常的CommitLog文件，理论上好像也能用来着

**Reply by domyway** [reply]
> 看代码确实存在这个问题，并且最新版本也没看到有解决。想问下你们测试的时候这个问题会影响使用吗？看逻辑MappedFileQueue应该会自动跳过这种异常的CommitLog文件，理论上好像也能用来着

1. 看起来进程正常跑，只是预创建了一个空文件，这个空文件不会被使用到
2. 进程在正常shutdown的时候会auto delete空文件
3. 进程异常crash，那么这个问题就有风险了
